source/widgets/ProjectStatus/js/configuration.js

```diff
@@ -29,13 +30,25 @@ function OctopusStatusWidgetConfiguration() {
                 return customSettings;
             };
 
-            var fetchDataSourceContent = function (queryUri, token, name, spaceId) {
-                let query;
+            var fetchDataSourceContent = function (queryUri, token, name, spaceId, partialNameFilter) {
+                let parameters = [];
                 if (spaceId) {
-                    query = '{"dataSourceDetails": {"dataSourceName":"' + name + '", "parameters": {"SpaceId":"' + spaceId + '"} } }';
+                    parameters.push('"SpaceId":"' + spaceId + '"');
+                }
+
+                // We check explicitly for `null` in this case. There are cases where we need to pass an empty string through to our endpointUrl.
+                if (partialNameFilter != null) {
+                    parameters.push('"PartialNameFilter":"' + partialNameFilter + '"');
+                }
+
+                let query;
+                if (parameters.length > 0) {
+                    let parametersCsv = "{" + parameters.join(", ") + "}";
```

Comment:
y'know, I know we paired on this, but looking at it again, couldn't this whole query just be an object that we add properties to and then call `JSON.stringify()`?

Reply:
Good call, and yes we could. In this case, the `query` further down is also done using string manipulation, so I've opted to keep with the current approach.

I do see this whole thing getting a rewrite in the near future (to support those lovely new libraries also being discussed here), so we could also introduce typing at that point and put some interfaces around this. For now, I'll add a TODO discussing the plan to add TS and we can revisit this (and anywhere else we're doing this sort of thing) so we can add typing and JSON.stringify() at that point.